Script started on 2025-08-21 04:02:31-04:00 [COMMAND="set -x && pytest tests/" TERM="screen.xterm-256color" TTY="/dev/pty5" COLUMNS="121" LINES="40"]
+ pytest tests/
................................s..F.................F..                                                          [100%]
======================================================= FAILURES ========================================================
_____________________________________ TestCygwinDetection.test_is_cygwin_detection ______________________________________
self = <tests.test_pth_functionality.TestCygwinDetection testMethod=test_is_cygwin_detection>
mock_exists = <MagicMock name='exists' id='123145277023520'>
mock_system = <MagicMock name='system' id='123145276798528'>
    @patch('setup.platform.system')
    @patch('setup.os.path.exists')
    def test_is_cygwin_detection(self, mock_exists, mock_system):
        """Test various Cygwin detection scenarios."""
        # Import here to ensure we're testing the setup module's is_cygwin
        from setup import is_cygwin
    
        # Test positive detection
        mock_system.return_value = 'CYGWIN_NT-10.0'
        self.assertTrue(is_cygwin())
    
        # Test negative detection
        mock_system.return_value = 'Windows'
        mock_exists.return_value = False
>       self.assertFalse(is_cygwin())
E       AssertionError: True is not false
tests/test_pth_functionality.py:391: AssertionError
_____________________________________ TestErrorHandling.test_access_denied_handling _____________________________________
self = <tests.test_unit.TestErrorHandling testMethod=test_access_denied_handling>
mock_exists = <MagicMock name='exists' id='123145278427200'>, mock_open = <MagicMock name='open' id='123145276386704'>
    @patch('builtins.open', side_effect=PermissionError("Access denied"))
    @patch('os.path.exists', return_value=True)
    def test_access_denied_handling(self, mock_exists, mock_open):
        """Test handling of permission errors."""
        proc = psutil.Process(1)
        with self.assertRaises(psutil.AccessDenied):
>           proc.name()
E           AssertionError: AccessDenied not raised
tests/test_unit.py:220: AssertionError
=================================================== warnings summary ====================================================
../../../../usr/lib/python3.9/site-packages/setuptools/command/easy_install.py:41
  /usr/lib/python3.9/site-packages/setuptools/command/easy_install.py:41: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources
../../../../usr/lib/python3.9/site-packages/pkg_resources/__init__.py:3142
  /usr/lib/python3.9/site-packages/pkg_resources/__init__.py:3142: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)
tests/test_pth_functionality.py::TestInstallationIntegration::test_cygwin_install_command_calls_pth_creation
tests/test_pth_functionality.py::TestInstallationIntegration::test_cygwin_uninstall_command_calls_pth_removal
  /usr/lib/python3.9/site-packages/setuptools/_distutils/cmd.py:79: SetuptoolsDeprecationWarning: setup.py install is deprecated.
  !!
  
          ********************************************************************************
          Please avoid running ``setup.py`` directly.
          Instead, use pypa/build, pypa/installer or other
          standards-based tools.
  
          See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
          ********************************************************************************
  
  !!
    self.initialize_options()
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================ short test summary info ================================================
SKIPPED [1] tests/test_pth_functionality.py:297: psutil not available (may not be installed)
FAILED tests/test_pth_functionality.py::TestCygwinDetection::test_is_cygwin_detection - AssertionError: True is not false
FAILED tests/test_unit.py::TestErrorHandling::test_access_denied_handling - AssertionError: AccessDenied not raised
2 failed, 53 passed, 1 skipped, 4 warnings in 8.10s
Script done on 2025-08-21 04:02:40-04:00 [COMMAND_EXIT_CODE="1"]

